### PR TITLE
LPS-204874 Adapting test cases to current validation in the Object layer

### PR DESCRIPTION
**Pull request by magjed4289**
https://issues.liferay.com/browse/LPS-204874

### CODE REVIEW

**Comment by magjed4289** [comment]
ci:help

**Comment by magjed4289** [comment]
ci:test:headless-builder

**Comment by magjed4289** [comment]
ci:test:relevant

**Comment by magjed4289** [comment]
ci:stop:headless-builder

**Comment by ccorreagg** [comment]
So which is the current problem? The current tests are not properly designed? Did we change the validations?

**Comment by sergiojimcos** [comment]
hi @ccorreagg! As @magjed4289 and I detected yesterday, there weren't any previous validations from Object Infrastructure when an user uses an invalid related Object  Entry id,  like this:

```json
curl -X 'POST' 
  'http://localhost:8080/o/headless-builder/endpoints/' 
  -H 'accept: application/json' 
  -H 'Content-Type: application/json' 
  -H 'test@liferay.com:test' 
  -d '{ 
  "httpMethod": "get",
  "name": "test",
  "path": "/testpath",
  "retrieveType": "collection",
  "r_apiApplicationToAPIEndpoints_c_apiApplicationId": 123
}'
```
Previously, there weren't any validation that checks if the `r_apiApplicationToAPIEndpoints_c_apiApplicationId` which is 123 is a valid Object Entry. If not, it will throw a `NotSuchModelException`. So the aim of this PR is to update all the API Builder test cases and instead of use `Random.nextLong()` statements to throw API Builder validations, we will use a random Object Entry from a random Object Definition.

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java [hunk 1]
```diff
@@ -156,39 +131,6 @@ public void testCreateGetAPIEndpoint() throws Exception {
 			).toString(),
 			"headless-builder/applications", Http.Method.POST);
 
-		JSONAssert.assertEquals(
```
Comment (sergiojimcos) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java:159]: Why did you delete this test scenario? You can substitute `RandomTestUtil.nextLong()` with `_objectEntry.getId()`

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java [hunk 2]
```diff
@@ -57,6 +63,18 @@ public void test() throws Exception {
 			).toString(),
 			"headless-builder/applications", Http.Method.POST);
 
+		JSONObject apiApplicationPublishedJSONObject =
+			HTTPTestUtil.invokeToJSONObject(
+				JSONUtil.put(
+					"applicationStatus", "unpublished"
```
Comment (sergiojimcos) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java:69]: it should be `published` due to the name is `apiApplicationPublishedJSONObject`.

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java [hunk 3]
```diff
@@ -85,6 +103,18 @@ public void test() throws Exception {
 			"An API schema must be related to an API application.",
 			jsonObject.get("title"));
 
+		JSONObject apiSchemaJSONObject = HTTPTestUtil.invokeToJSONObject(
```
Comment (sergiojimcos) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java:106]: I think that you can reuse the apiSchema at line `168`, because they are the same. As well as you can delete `apiApplicationPublishedJSONObject`.

**Review by sergiojimcos — CHANGES_REQUESTED** [review]
Please review my comments and suggestions

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java (hunk 1 above)
Comment (magjed4289) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java:159]: I did it in another test, that's why this one seemed redundant 😅

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java (hunk 3 above)
Comment (magjed4289) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java:106]: will check that! thanks!

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java (hunk 1 above)
Comment (sergiojimcos) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java:159]: But be careful, because the error messages are not the same :smile:

**Comment by magjed4289** [comment]
ci:test:relevant

**Comment by magjed4289** [comment]
all [headless builder tests ](https://testray.liferay.com/home/-/testray/case_results/index?p_p_state=normal&orderByCol=status_sortable&orderByType=asc&testrayBuildId=2482472603&testrayComponentId=2391083629&testrayRunId=2482475387) are passing

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java (hunk 1 above)
Comment (magjed4289) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIEndpointRelevantObjectEntryModelListenerTest.java:159]: added it back, thanks for catching it! :)

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java [hunk 4]
```diff
@@ -118,6 +129,40 @@ public void test() throws Exception {
 			"An API schema must be an existing object definition.",
 			jsonObject.get("title"));
 
+		JSONAssert.assertEquals(
```
Comment (sergiojimcos) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java:132]: I think this test scenario should be in `APIEndpointRelevantObjectEntryModelListenerTest`, because you are using the `headless-builder/endpoint`, what do you think?

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java [hunk 5]
```diff
@@ -138,6 +183,23 @@ public void test() throws Exception {
...
+				"name", RandomTestUtil.randomString()
+			).put(
+				"r_apiApplicationToAPISchemas_c_apiApplicationId",
+				jsonObject.getLong("id")
+			).toString(),
+			"headless-builder/schemas", Http.Method.POST);
+
+		Assert.assertEquals("BAD_REQUEST", jsonObject.get("status"));
+		Assert.assertEquals(
+			"An API schema must be related to an API application.",
```
Comment (sergiojimcos) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java:200]: Try to sort all the tests depending on its error message, please, because if not BC will request you to do it. Sorry for not letting you to know it :pray:

**Review by sergiojimcos — CHANGES_REQUESTED** [review]
Just two comments and the PR will be ready

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java (hunk 5 above)
Comment (magjed4289) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java:200]: beter late then sorry :D Thanks for letting me know, will sort this now!

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java (hunk 4 above)
Comment (magjed4289) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISchemaRelevantObjectEntryModelListenerTest.java:132]: will analyse now and move if needed!

**Comment by magjed4289** [comment]
ci:test:headless-builder

**Comment by magjed4289** [comment]
ci:test:relevant

**Review by sergiojimcos — APPROVED** [review]
LGTM!! :clap: :clap: Great job!

**Comment by magjed4289** [comment]
ci:forward
